src/main/java/com/onionshop/Project.java

```diff
@@ -76,31 +82,62 @@ public String getPath() {
     /**
      * Sets new path
      *
-     * @param newPath: the new path to which this.path will be set
+     * @param newPath the new path to which this.path will be set
      */
-    public void updatePath(String newPath) {
-        path = newPath;
-        //TODO (optional): add a checker to see if the path is valid and writable
+    public void updatePath(String newPath) throws Exception {
+        if (OnionFileLoader.isDirectoryValid(newPath)) {
+            path = newPath;
+        } else {
+            throw new Exception("Invalid path was given: " + newPath);
+        }
+
     }
 
+    /**
+     * Adds a new custom colour to the colour palette
+     *
+     * @param newColour the new Colour instance to be added
+     */
     public void addColour(Colour newColour) {
-        Colour[] newCP = new Colour[colourPalette.length + 1];
-        for (int i = 0; i < colourPalette.length; i++) {
-            newCP[i] = colourPalette[i];
```

Comment:
Should Project handle these, or should we have a separate ColourPalette class handle these? Just a thought, if its not inconvenient. So we can make sure were adhering to the single responsibility principle